commands/databases.go

```diff
@@ -511,13 +513,57 @@ func RunDatabaseResize(c *CmdConfig) error {
 	}
 
 	id := c.Args[0]
+	dbs := c.Databases()
 
 	r, err := buildDatabaseResizeRequestFromArgs(c)
 	if err != nil {
 		return err
 	}
 
-	return c.Databases().Resize(id, r)
+	// Resize the database
+	err = dbs.Resize(id, r)
+	if err != nil {
+		return err
+	}
+
+	// Retrieve the database object after resize (to check status or further actions)
+	db, err := dbs.Get(id)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve the database after resize: %v", err)
+	}
+
+	// Check if the --wait flag was passed
+	wait, err := c.Doit.GetBool(c.NS, doctl.ArgCommandWait)
+	if err != nil {
+		return err
+	}
+
+	if wait {
+		connection := db.Connection
+
+		notice("Database resizing is in progress, waiting for database to be online")
+
+		err := waitForDatabaseReady(dbs, id)
+		if err != nil {
+			return fmt.Errorf(
+				"database couldn't enter the `online` state after resizing: %v",
+				err,
+			)
+		}
+
+		db, err = dbs.Get(id)
+		if err != nil {
+			return fmt.Errorf(
+				"failed to retrieve the resized database: %v",
+				err,
+			)
+		}
+		db.Connection = connection
+	}
+
+	notice("Database resized successfully")
+
+	return displayDatabases(c, false, *db)
```

Comment:
```suggestion
		notice("Database resized successfully")
	}

	return nil
```

Let's also move the notice inside the wait block. That way the current behavior remains the same if you do not pass the `--wait` flag.